### PR TITLE
Fixes #30369 - Added condition to check .snar files

### PR DESCRIPTION
**Pull request by patilsuraj767**

### CODE REVIEW

File: definitions/procedures/backup/prepare_directory.rb [hunk 1]
```diff
@@ -22,8 +23,14 @@ def run
 
       FileUtils.rm(Dir.glob(File.join(@backup_dir, '.*.snar'))) if @preserve_dir
       if @incremental_dir
-        FileUtils.cp(Dir.glob(File.join(@incremental_dir, '.*.snar')), @backup_dir)
+        snar_files = Dir.glob(File.join(@incremental_dir, '.*.snar'))
+        if snar_files.empty?
```
Comment (upadhyeammit) [inline definitions/procedures/backup/prepare_directory.rb:27]: You can combine `!snar_files.empty?` condition with parent if condition ?  As we should take incremental backup only when snar_files are available?

Comment (patilsuraj767) [reply definitions/procedures/backup/prepare_directory.rb:27]: @upadhyeammit 
In `snar_files = Dir.glob(File.join(@incremental_dir, '.*.snar'))` we use `@incremental_dir`. So I think it is good to first check if `@incremental_dir` exist. If we shift the `!snar_files.empty?` to parent then foreman-maintain will every time try to check the snar files without checking if  `@incremental_dir` exist. 
Let me know your thoughts on above.

Comment (upadhyeammit) [reply definitions/procedures/backup/prepare_directory.rb:27]: Agree with you !

File: definitions/procedures/backup/prepare_directory.rb [hunk 2]
```diff
@@ -22,8 +23,14 @@ def run
 
       FileUtils.rm(Dir.glob(File.join(@backup_dir, '.*.snar'))) if @preserve_dir
       if @incremental_dir
```
Comment (kgaikwad) [inline definitions/procedures/backup/prepare_directory.rb:25]:
```suggestion
      if @incremental_dir && !(snar_files = Dir.glob(File.join(@incremental_dir, '.*.snar'))).empty?
```

File: definitions/procedures/backup/prepare_directory.rb [hunk 3]
```diff
@@ -21,8 +21,10 @@ def run
       end
 
       FileUtils.rm(Dir.glob(File.join(@backup_dir, '.*.snar'))) if @preserve_dir
-      if @incremental_dir
-        FileUtils.cp(Dir.glob(File.join(@incremental_dir, '.*.snar')), @backup_dir)
+      if @incremental_dir && !(snar_files = Dir.glob(File.join(@incremental_dir, '.*.snar'))).empty?
+        FileUtils.cp(snar_files, @backup_dir)
+      else
+        raise 'No .snar files found in previous backup directory'
```
Comment (kgaikwad) [inline definitions/procedures/backup/prepare_directory.rb:27]: As there are two possibilities, I think we should change error message. Something like -
```suggestion
        raise "May be either incremental backup directory or details such as .snar files under this directory is missing. Could you please provide a valid previous backup directory."
```
@upadhyeammit, any suggestions on message wording?

Comment (upadhyeammit) [reply definitions/procedures/backup/prepare_directory.rb:27]: Maybe:

"Either #{@incremental_dir} or #{@incremental_dir}/*.snar files unavailable. Provide valid previous backup directory"

**Pull request merged**